components/main_menu/main_menu.jsx

```diff
@@ -281,17 +283,12 @@ class MainMenu extends React.PureComponent {
                     {pluginItems}
                 </Menu.Group>
                 <Menu.Group>
-                    <TeamPermissionGate
-                        teamId={this.props.teamId}
-                        permissions={[Permissions.MANAGE_SLASH_COMMANDS, Permissions.MANAGE_OAUTH, Permissions.MANAGE_INCOMING_WEBHOOKS, Permissions.MANAGE_OUTGOING_WEBHOOKS]}
-                    >
-                        <Menu.ItemLink
-                            id='integrations'
-                            show={!this.props.mobile && (this.props.enableIncomingWebhooks || this.props.enableOutgoingWebhooks || this.props.enableCommands || this.props.enableOAuthServiceProvider)}
-                            to={'/' + this.props.teamName + '/integrations'}
-                            text={formatMessage({id: 'navbar_dropdown.integrations', defaultMessage: 'Integrations'})}
-                        />
-                    </TeamPermissionGate>
+                    <Menu.ItemLink
+                        id='integrations'
+                        show={!this.props.mobile && this.props.canManageIntegrations && (this.props.enableIncomingWebhooks || this.props.enableOutgoingWebhooks || this.props.enableCommands || this.props.enableOAuthServiceProvider || this.props.canManageSystemBots)}
```

Comment:
Is checking here for `this.props.canManageSystemBots` needed if it's already included in `this.props.canManageIntegrations` check?

Reply:
Unfortunately, yes. There's effectively three booleans at play: `canManageTeamIntegrations`, `canManageSystemBots`, and "`someTeamIntegrationEnabled`" (i.e. `this.props.enableIncomingWebhooks || this.props.enableOutgoingWebhooks || this.props.enableCommands || this.props.enableOAuthServiceProvider`)

`canManageSystemBots` does double duty as a permission and enablement check, but we test it again because it's possible that a user has permission to manage integrations but not bots (so `canManageIntegrations` is true), but no integrations are actually enabled. So we suppress the menu item in that case.

This is decidedly not clear, so I've pulled this out into a variable that hopefully helps.

Reply:
Thanks for that clarification! Wondering why we don't have a similar setting to enable bots?